docs/DeveloperGuide.md

```diff
@@ -1,22 +1,264 @@
+#### Architecture 
+
+Main components of the architecture:
+
+`Main` (consisting of class `JavaNinja`) is in charge of the app launch and shut down. 
+1. At app launch, it initialises the components in the correct order and connects them with each other
+2. At shut down, it shuts down the other components and invokes cleanup methods 
+
+The bulk of the app's work is done by the respective components: 
+1. `Cli`: It handles the user interface of the app 
+2. `Parser`: The command executor 
+3. `QuizManager`: Holds the data of the App in memory
+4. `Storage`: Reads data from, and writes data to, the hard disk.
+
+`Commons` represent a collection of classes used by other components.
+
+How the architecture components interact with each other: 
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `add Flashcard /q what's 2+3 /a 5`
+
+![image](https://github.com/user-attachments/assets/1dd443de-b538-4fb8-b4d1-3bff4643b3dd)
+
+### QuizManager Class Implementation
+The `QuizManager` class orchestrates the main operations of the quiz application. It manages topics, quizzes, past results, and handles the loading and saving of data. This class serves as the central controller that interacts with other classes like `Topic`, `Quiz`, and `Storage`.
+
+#### **Feature Overview**
+The `QuizManager` class is designed to provide the following key functionalities: 
+
+1. **Manage Topic and Questions**:
+    - Load and save questions by topic, allowing easy organization of quiz content. 
+    - Support for multiple question types, such as multiple-choice, true/false, flashcards, and fill-in-the-blank.
+
+2. **Quiz Execution**:
+    - Enable users to start quizzes on specific topics, with time limits and question limits. 
+    - Track the user’s score and display feedback upon quiz completion.
+
+3. **Persistence of Quiz Data**:
+    - Save quiz questions and past quiz results to files, ensuring data is preserved across sessions.
+    - Load saved quiz data from files when the application starts.
+
+4. **Review Past Results**:
+    - Allow users to review their scores and comments from previous quiz attempts, providing a way to track their progress.
+
+
+#### **Implementation Details**
+
+Implementation Details
+
+1. **Attributes**:
+   - Constants:
+     - `QUESTIONS_FILE_PATH`: Specifies the path to the file that stores quiz questions.
+     - `RESULTS_FILE_PATH`: Specifies the path to the file that stores past quiz results.
+   - Core Components:
+     - `List<Topic> topics`: Stores all available quiz topics.
+     - `Quiz currentQuiz`: Represents the quiz currently in progress.
+     - `List<String> pastResults`: Maintains a list of past quiz results, with each result stored as a formatted string.
+   - Data Storage:
+     - `Storage results`: Manages loading and saving of past quiz results.
+     - `Storage questions`: Manages loading and saving of quiz questions.
```

Comment:
Maybe consider listing the attributes constant throughout? As some has variable type but some do not